### PR TITLE
feature: add reusable form components and settings tabs

### PR DESCRIPTION
**Pull request by rpollock03**
Settings page, tabs and reusable components. 
Screenshot 2023-07-23 at 10 27 31

Added reusable form components including:
- toggle nav component, 'TabSelect'. Tab components are passed as children to TabSelect, and then mapped.
- reusable dropdown (select) component with menu options passed in as props.
- reusable modal component

Also added all settings tabs, using reusable form components and keeping consistent with the 'New Quest' form.

**Reusable toggle component:**
Screenshot 2023-07-08 at 10 24 02

**Reusable dropdown component:**
Screenshot 2023-07-18 at 21 14 15

**Reusable modal component:**
Screenshot 2023-07-22 at 12 35 55

**Account Tab form:**
Screenshot 2023-07-22 at 22 26 52

**Notifications Tab form:**
Screenshot 2023-07-22 at 10 17 59

**Tested to make sure information is correctly updated to, or retrieved from firebase:**
Screenshot 2023-07-22 at 10 15 30

**Security Tab form: **
Screenshot 2023-07-22 at 22 16 28

**Location Tab: **
Screenshot 2023-07-23 at 10 22 55

**Tax Tab:**
Screenshot 2023-07-22 at 23 14 06

**Balances Tab:**
Screenshot 2023-07-23 at 13 05 30

### CODE REVIEW

**Comment by OliwierOST** [comment]
I think we could make the toggle nav reusable as it appears in multiple places throughout the website, e.g Quests Hub (Client) on Figma. Let's pass the navigation items as props to the component.

**Comment by rpollock03** [comment]
> I think we could make the toggle nav reusable as it appears in multiple places throughout the website, e.g Quests Hub (Client) on Figma. Let's pass the navigation items as props to the component.

Thanks @OliwierOST, I have made this into a reusable component.

File: components/FormField/index.tsx [hunk 1]
```diff
@@ -6,16 +6,20 @@ interface FormFieldProps {
...
   setFieldValue?: SetFieldValue
+  changeState?: (a: string) => void
 }
 
 export function FormField({
   rows = 1,
   placeholder = "",
   value = "",
   name = "",
+  type = "",
```
Comment (rpollock03) [inline components/FormField/index.tsx:20]: This is necessary to allow form fields to be password fields for example.

**Review by oliwierost — CHANGES_REQUESTED** [review]
Some minor changes. Other than that very good work! Keep it up!

File: components/FormDropdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        IconComponent={icon}
+        inputProps={{
+          sx: {
+            borderRadius: 2,
+            width: "20rem",
+          },
+        }}
+      >
+        {options.map((option, index) => {
+          return (
```
[the remark this answers is not in the document]
Comment (rpollock03) [reply components/FormDropdown/index.tsx:36]: done

File: components/FormDropdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+interface IMenuOption {
+  value: string
+  label: string
+}
+
+interface FormDropdownProps {
+  value?: string
+  options?: IMenuOption[] | string[]
+  changeState?: (a: string) => void
+  icon?: any
```
[the remark this answers is not in the document]
Comment (rpollock03) [reply components/FormDropdown/index.tsx:12]: This doesn't work, trying to find alternatives

File: components/Settings/accountTab.tsx [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+          severity: "success",
+        })
+      })
+      .catch((error) => {
+        setSnackbar({
+          open: true,
+          message: "User deletion failed!",
+          severity: "error",
+        })
+      })
```
[the remark this answers is not in the document]
Comment (rpollock03) [reply components/Settings/accountTab.tsx:91]: done

File: components/Settings/securityTab.tsx [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+        <Formik
+          initialValues={{
+            question1: "",
+            answer1: "",
+            question2: "",
+            answer2: "",
+            question3: "",
+            answer3: "",
+          }}
+          onSubmit={async (values) => {
```
[the remark this answers is not in the document]
Comment (rpollock03) [reply components/Settings/securityTab.tsx:263]: done

File: components/PrimaryButton/index.tsx [hunk 6]
```diff
@@ -20,6 +21,7 @@ export function PrimaryButton({
   disabled = false,
   type = "button",
   sx = {},
+  form,
```
[the remark this answers is not in the document]
Comment (rpollock03) [reply components/PrimaryButton/index.tsx:24]: done

File: components/FormField/index.tsx [hunk 7]
```diff
@@ -6,16 +6,20 @@ interface FormFieldProps {
...
   setFieldValue,
+  changeState,
```
[the remark this answers is not in the document]
Comment (rpollock03) [reply components/FormField/index.tsx:22]: done

File: components/Settings/balancesTab.tsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+  return (
+    <Stack spacing={2}>
+      <Stack direction="row">
+        <FormInputLabel label="Available Balance" />
+        <Typography sx={{ fontWeight: "bold" }} variant="formBody">
+          ${balance}USD
+        </Typography>
+      </Stack>
+      <Stack direction={"row"} justifyContent="flex-end" mt={1} spacing={1}>
```
[the remark this answers is not in the document]
Comment (rpollock03) [reply components/Settings/balancesTab.tsx:33]: fixed

File: components/FormDropdown/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (rpollock03) [reply components/FormDropdown/index.tsx:12]: fixed

File: components/Settings/accountTab.tsx [hunk 9]
```diff
@@ -0,0 +1,394 @@
...
+  }
+
+  const handleSaveDiscipline = async () => {
+    try {
+      await updateDoc(heroRef, {
+        workType: workType,
+        subWorkType: subWorkType,
+      })
+      {
+        /* TODO - add arrayUnion to save tags */
```
[the remark this answers is not in the document]
Comment (rpollock03) [reply components/Settings/accountTab.tsx:125]: done

**Comment by rpollock03** [comment]
all requested changes made @oliwierost , should be good to merge!

**Review by oliwierost — APPROVED** [review]
Very good! Keep an eye on little stylistic issues, e.g. when passing prop dont't do `label={"Name"}`, instead do ` label="Name"`. Also a lot of components let you use padding and marigin directly as a prop like this: `px={4}`. We'd rather wan't to use that approach instead of `sx={{ px: 4 }}`. Other than that a lot of improvements and overall very nice!

**Pull request merged**